diff -rup a/sys/fbdev/gstfbdevsink.c b/sys/fbdev/gstfbdevsink.c
--- a/sys/fbdev/gstfbdevsink.c	2011-12-30 19:24:33.000000000 +0100
+++ b/sys/fbdev/gstfbdevsink.c	2015-04-15 12:34:49.924673352 +0200
@@ -43,6 +43,8 @@ enum
 {
   ARG_0,
   ARG_DEVICE,
+  ARG_XOFFSET,
+  ARG_YOFFSET,
 };
 
 static void gst_fbdevsink_base_init (gpointer g_class);
@@ -92,7 +94,7 @@ gst_fbdevsink_base_init (gpointer g_clas
 
   gst_element_class_set_details_simple (element_class, "fbdev video sink",
       "Sink/Video",
-      "A linux framebuffer videosink", "Sean D'Epagnier <sean@depagnier.com>");
+      "A linux framebuffer videosink (DERMALOG mod)", "Sean D'Epagnier <sean@depagnier.com>,Andrej Kostrov <andrej.kostrov@dermalog.com>");
   gst_element_class_add_static_pad_template (element_class,
       &sink_template);
 }
@@ -199,6 +201,7 @@ gst_fbdevsink_setcaps (GstBaseSink * bsi
 {
   GstFBDEVSink *fbdevsink;
   GstStructure *structure;
+  int xoffset_max, yoffset_max;
   const GValue *fps;
 
   fbdevsink = GST_FBDEVSINK (bsink);
@@ -215,11 +218,29 @@ gst_fbdevsink_setcaps (GstBaseSink * bsi
   /* calculate centering and scanlengths for the video */
   fbdevsink->bytespp = fbdevsink->fixinfo.line_length / fbdevsink->varinfo.xres;
 
-  fbdevsink->cx = ((int) fbdevsink->varinfo.xres - fbdevsink->width) / 2;
+  xoffset_max =  fbdevsink->varinfo.xres - fbdevsink->width ;
+  if (fbdevsink->xoffset < 0) // center
+    fbdevsink->cx = xoffset_max / 2;
+  else{
+    if (fbdevsink->xoffset > xoffset_max)
+      fbdevsink->cx = xoffset_max;
+    else
+      fbdevsink->cx = fbdevsink->xoffset;
+  }
+
   if (fbdevsink->cx < 0)
     fbdevsink->cx = 0;
 
-  fbdevsink->cy = ((int) fbdevsink->varinfo.yres - fbdevsink->height) / 2;
+  yoffset_max = fbdevsink->varinfo.yres - fbdevsink->height;
+  if (fbdevsink->yoffset < 0)
+    fbdevsink->cy = yoffset_max / 2;
+  else{
+    if (fbdevsink->yoffset > yoffset_max)
+      fbdevsink->cy = yoffset_max;
+    else
+      fbdevsink->cy = fbdevsink->yoffset;
+  }
+
   if (fbdevsink->cy < 0)
     fbdevsink->cy = 0;
 
@@ -322,6 +343,14 @@ gst_fbdevsink_set_property (GObject * ob
       fbdevsink->device = g_value_dup_string (value);
       break;
     }
+    case ARG_XOFFSET:{
+      fbdevsink->xoffset = g_value_get_int(value);
+      break;
+    }
+    case ARG_YOFFSET:{
+      fbdevsink->yoffset = g_value_get_int(value);
+      break;
+    }
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
       break;
@@ -342,6 +371,14 @@ gst_fbdevsink_get_property (GObject * ob
       g_value_set_string (value, fbdevsink->device);
       break;
     }
+    case ARG_XOFFSET:{
+      g_value_set_int (value, fbdevsink->xoffset);
+      break;
+    }
+    case ARG_YOFFSET:{
+      g_value_set_int (value, fbdevsink->yoffset);
+      break;
+    }
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
       break;
@@ -398,6 +435,14 @@ gst_fbdevsink_class_init (GstFBDEVSinkCl
       g_param_spec_string ("device", "device",
           "The framebuffer device eg: /dev/fb0", NULL, G_PARAM_READWRITE));
 
+  g_object_class_install_property (G_OBJECT_CLASS (klass), ARG_XOFFSET,
+      g_param_spec_int ("xoffset", "xoffset", 
+          "The horizontal offset. xoffset=-1 center.", -1, G_MAXINT, -1, G_PARAM_READWRITE));
+  
+  g_object_class_install_property (G_OBJECT_CLASS (klass), ARG_YOFFSET,
+      g_param_spec_int ("yoffset", "yoffset", 
+          "The vertical offset. yoffset=-1 center.", -1, G_MAXINT, -1, G_PARAM_READWRITE));
+
   gstvs_class->set_caps = GST_DEBUG_FUNCPTR (gst_fbdevsink_setcaps);
   gstvs_class->get_caps = GST_DEBUG_FUNCPTR (gst_fbdevsink_getcaps);
   gstvs_class->get_times = GST_DEBUG_FUNCPTR (gst_fbdevsink_get_times);
@@ -446,5 +491,5 @@ gst_fbdevsink_get_type (void)
 GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
     GST_VERSION_MINOR,
     "fbdevsink",
-    "linux framebuffer video sink",
+    "linux framebuffer video sink (DERMALOG mod)",
     plugin_init, VERSION, GST_LICENSE, GST_PACKAGE_NAME, GST_PACKAGE_ORIGIN)
diff -rup a/sys/fbdev/gstfbdevsink.h b/sys/fbdev/gstfbdevsink.h
--- a/sys/fbdev/gstfbdevsink.h	2011-12-11 19:49:14.000000000 +0100
+++ b/sys/fbdev/gstfbdevsink.h	2015-04-15 12:34:49.924673352 +0200
@@ -59,7 +59,7 @@ struct _GstFBDEVSink {
   unsigned char *framebuffer;
 
   char *device;
-
+  int xoffset, yoffset;
   int width, height;
   int cx, cy, linelen, lines, bytespp;
 
